services/src/main/java/org/keycloak/url/HostnameV2ProviderFactory.java

```diff
@@ -78,17 +73,36 @@ public void init(Config.Scope config) {
             throw new IllegalArgumentException("hostname-backchannel-dynamic must be set to false if hostname is not provided as full URL");
         }
     }
+    
+    private void validateAndSetHostname(String hostname) {
+        URI result;
+        try {
+            result = URI.create("http://"+hostname);
+        }
+        catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException(INVALID_HOSTNAME, e);
+        }
+        if (result.getHost() == null || !result.getHost().equals(hostname)) {
+            throw new IllegalArgumentException(INVALID_HOSTNAME);
+        }
+        this.hostname = hostname;
+    }
```

Comment:
I believe, something like `foobar.com/path` would also pass for a plain hostname and it would be evaluated as `hostname=foobar.com`. I believe the validation should fail in cases like this.

Reply:
This case is covered: https://github.com/keycloak/keycloak/pull/32659/files#diff-173c36cf91b186489908e31566b3d2fb199c41bb4b898f6787796ae194fb083fR42

The check for getHost equality with the hostname makes sure that nothing else can appear.

Reply:
Ah, right, sorry for the noise.